### PR TITLE
[THOG-709] - Fix recover

### PR DESCRIPTION
**Pull request by ahrav**

### CODE REVIEW

File: pkg/engine/engine.go [hunk 1]
```diff
@@ -186,16 +181,21 @@ func (e *Engine) detectorWorker(ctx context.Context) {
...
 						continue
 					}
-					ctx, cancel := context.WithTimeout(ctx, time.Second*10)
-					defer cancel()
-					results, err := detector.FromData(ctx, verify, decoded.Data)
+
+					results, err := func() ([]detectors.Result, error) {
+						ctx, cancel := context.WithTimeout(ctx, time.Second*10)
+						defer cancel()
+						defer common.Recover(ctx)
```
Comment (mcastorina) [inline pkg/engine/engine.go:188]: Do you know if the order matters?

**Review by mcastorina — APPROVED** [review]

File: pkg/engine/engine.go (hunk 1 above)
Comment (ahrav) [reply pkg/engine/engine.go:188]: Yea i think we want recover, then cancel. So with defer it's LIFO so I think this works.

Comment (ahrav) [reply pkg/engine/engine.go:188]: hm... or maybe we want that flipped.

Comment (ahrav) [reply pkg/engine/engine.go:188]: OKay yep cancel then recover looks correct.

**Pull request merged**